src/chunkserver/chunkserver_entry.cc

```diff
@@ -0,0 +1,1578 @@
+		}
+	} else {
+		std::vector<uint8_t> buffer;
+		eptr->messageSerializer->serializeCstoclReadStatus(
+		    buffer, eptr->chunkId, status);
+		eptr->createAttachedPacket(buffer);
+		if (eptr->isChunkOpen) {
+			job_close(eptr->workerJobPool, nullptr, nullptr, eptr->chunkId,
+			          eptr->chunkType);
+			eptr->isChunkOpen = 0;
+		}
+		// after sending status even if there was an error it's possible to
+		eptr->state = State::Idle;
+		// receive new requests on the same connection
+		LOG_AVG_STOP(readOperationTimer);
+	}
+}
+
+void ChunkserverEntry::sendFinished() {
+	TRACETHIS();
+	todoReadCounter--;
+	if (todoReadCounter == 0) {
+		readContinue();
+	}
+}
+
+void ChunkserverEntry::readContinue() {
+	TRACETHIS2(offset, size);
+
+	if (readPacket) {
+		attachPacket(std::move(readPacket));
+		readPacket.reset();
+		todoReadCounter++;
+	}
+	if (size == 0) {  // everything has been read
+		std::vector<uint8_t> buffer;
+		messageSerializer->serializeCstoclReadStatus(
+		    buffer, chunkId, SAUNAFS_STATUS_OK);
+		createAttachedPacket(buffer);
+		sassert(isChunkOpen);
+
+		job_close(workerJobPool, nullptr, nullptr, chunkId,
+		          chunkType);
+		isChunkOpen = 0;
+		// no error - do not disconnect - go direct to the IDLE state, ready for
+		// requests on the same connection
+		state = State::Idle;
+		LOG_AVG_STOP(readOperationTimer);
+	} else {
+		const uint32_t totalRequestSize = size;
+		const uint32_t thisPartOffset = offset % SFSBLOCKSIZE;
+		const uint32_t thisPartSize = std::min<uint32_t>(
+				totalRequestSize, SFSBLOCKSIZE - thisPartOffset);
+		const uint16_t totalRequestBlocks =
+		    (totalRequestSize + thisPartOffset + SFSBLOCKSIZE - 1) /
+		    SFSBLOCKSIZE;
+
+		std::vector<uint8_t> readDataPrefix;
+		messageSerializer->serializePrefixOfCstoclReadData(
+		    readDataPrefix, chunkId, offset, thisPartSize);
+		auto packet = createDetachedPacketWithOutputBuffer(readDataPrefix);
+		if (packet == nullptr) {
+			state = State::Close;
+			return;
+		}
+		readPacket = std::move(packet);
+
+		uint32_t readAheadBlocks = 0;
+		uint32_t maxReadBehindBlocks = 0;
+
+		if (!static_cast<bool>(isChunkOpen)) {
+			if (gHDDReadAhead.blocksToBeReadAhead() > 0) {
+				readAheadBlocks =
+				    totalRequestBlocks + gHDDReadAhead.blocksToBeReadAhead();
+			}
+			// Try not to influence slow streams to much:
+			maxReadBehindBlocks = std::min(totalRequestBlocks,
+					gHDDReadAhead.maxBlocksToBeReadBehind());
+		}
+
+		readJobId = job_read(workerJobPool, readFinishedCallback, this, chunkId,
+		                     chunkVersion, chunkType, offset, thisPartSize,
+		                     maxReadBehindBlocks, readAheadBlocks,
+		                     readPacket->outputBuffer.get(), !isChunkOpen);
+		if (readJobId == 0) {
+			state = State::Close;
+			return;
+		}
+
+		todoReadCounter++;
+		offset += thisPartSize;
+		size -= thisPartSize;
+	}
+}
+
+void ChunkserverEntry::ping(const uint8_t *data, PacketHeader::Length length) {
+	if (length != 4) {
```

Comment:
Same here on considering naming the value 4 here for better readability after refactoring.